serverv2/core/appmanager/app.go

```diff
@@ -0,0 +1,54 @@
+package appmanager
+
+import (
+	"context"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/serverv2/core/event"
+	"github.com/cosmos/cosmos-sdk/serverv2/core/transaction"
+	"github.com/cosmos/cosmos-sdk/serverv2/core/validator"
+)
+
+type App[T transaction.Tx] interface {
+	ChainID() string
+	AppVersion() (uint64, error)
+
+	InitChain(context.Context, RequestInitChain) (ResponseInitChain, error)
+	DeliverBlock(context.Context, RequestDeliverBlock[T]) (ResponseDeliverBlock, error)
+}
+
+type RequestDeliverBlock[T transaction.Tx] struct {
+	Height int64
+	Time   time.Time
+	Hash   []byte
+
```

Comment:
I think I would add one more field here which could be called `ConsensusMessages` this thing could contain messages that the consensus wants to execute before the Txs, they are not validated, don't have ante handler etc. 

EG, this could be `cometbft.MsgSetEvidence` and could be handled by the cometbft module which is a consensus specific module. 

So the cometBFT server when delivering a block could populate these messages.

so